test/common/performance/receiver/id_extractor.go

```diff
@@ -0,0 +1,29 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package receiver
+
+import cloudevents "github.com/cloudevents/sdk-go"
+
+// IdExtractor is used by the receiver to establish the id of the received event
+type IdExtractor func(event cloudevents.Event) string
+
+// EventIdExtractor uses
+//  event.ID()
+// to extract the event id
```

Comment:
This is very self explanatory and doesn't need clarification in my opinion. What I meant is, could you explain in a short sentence _why_ we have those XExtractor types? Something along the lines of
> The implementation needs to be swappable because in some contexts (e.g. in sources tests) the ID may be embedded in the payload.

Reply:
PS: that comment could explain both Extractor, that's why I said a single `extractor.go` file would make sense IMO.

Reply:
I would prefer to add this comment in the [`Receiver` documentation](https://github.com/knative/eventing/pull/1979/files#diff-03dfc3d293a224e1a509cee40bb8b56cR37) because it's about the receiver behaviour, not about this type.

Reply:
Or there yes. We just need _a_ way to clarify it. Without you answering my question, I would have never figured out why you created types just to do `event.Foo()`

Reply:
Like this https://github.com/knative/eventing/pull/1979/commits/fb572100f43c9b99a168c5889e52694fca8cdf8b

Reply:
Sweet, now I understand :D Cheers!